File: TypeScript/Practice/LeetCode/Easy/Count Good Triplets.ts
// https://leetcode.com/problems/count-good-triplets/description/

function countGoodTriplets(arr: number[], a: number, b: number, c: number): number {
    let goodTripletsCounter: number = 0;

    for (let i = 0; i < arr.length; i++) {
        for (let j = i + 1; j < arr.length; j++) {
            for (let k = j + 1; k < arr.length; k++) {
                if (
                    Math.abs(arr[i] - arr[j]) <= a && 
                    Math.abs(arr[j] - arr[k]) <= b &&
                    Math.abs(arr[i] - arr[k]) <= c
                ) {
                    goodTripletsCounter++;
                }
            }
        }
    }

    return goodTripletsCounter;
};

File: C#/Practice/LeetCode/Easy/Count Good Triplets.cs
// https://leetcode.com/problems/count-good-triplets/description/

public class Solution {
    public int CountGoodTriplets(int[] arr, int a, int b, int c) {
        int goodTripletsCounter = 0;

        for (int i = 0; i < arr.Length; i++) {
            for (int j = i + 1; j < arr.Length; j++) {
                for (int k = j + 1; k < arr.Length; k++) {
                    if (
                        Math.Abs(arr[i] - arr[j]) <= a && 
                        Math.Abs(arr[j] - arr[k]) <= b &&
                        Math.Abs(arr[i] - arr[k]) <= c
                    ) {
                        goodTripletsCounter++;
                    }
                }
            }
        }

        return goodTripletsCounter;        
    }
}